pkg/provider/azure.go

```diff
@@ -373,6 +373,16 @@ func NewCloudFromConfigFile(configFilePath string, syncZones bool) (cloudprovide
 }
 
 func NewCloudFromSecret(clientBuilder cloudprovider.ControllerClientBuilder, secretName, secretNamespace, cloudConfigKey string) (cloudprovider.Interface, error) {
+	if secretName == "" {
+		secretName = consts.DefaultCloudProviderConfigSecName
+	}
+	if secretNamespace == "" {
+		secretName = consts.DefaultCloudProviderConfigSecNamespace
+	}
+	if cloudConfigKey == "" {
+		cloudConfigKey = consts.DefaultCloudProviderConfigSecKey
+	}
```

Comment:
also add a unit test for these default values?

Reply:
adding